### PR TITLE
Feature: define the starter workflows

### PR DESCRIPTION
**Pull request by laurentsimon**
User who install scorecards from the GH market place will see a default workflow automatically generated.
This PR defines the workflow that is used as template by GH market place. Note that we need to submit a PR at https://github.com/actions/starter-workflows for the changes to take effect. This PR only defines them locally in our repo so that we can discuss it and keep a copy of what it is.

Note: this PR follows the naming conventions of https://github.com/actions/starter-workflows for files and folders

### CODE REVIEW

File: starter-workflows/code-scanning/properties/scorecards.properties.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "name": "OSSF Scorecards supply-chain security scan",
+    "creator": "Open Source Security Foundation (OpenSSF) - https://github.com/ossf",
+    "description": "Scorecards is a static analysis tool to assess the security posture of your project",
+    "iconName": "scorecards",
+    "categories": [
+        "code-quality", "dependency-management", "testing",
```
Comment (laurentsimon) [inline starter-workflows/code-scanning/properties/scorecards.properties.json:7]: categories are from https://github.com/actions/starter-workflows#categories

File: starter-workflows/code-scanning/properties/scorecards.properties.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+        "supply-chain", "security", "scanning"
```
Comment (laurentsimon) [inline starter-workflows/code-scanning/properties/scorecards.properties.json:8]: we can also set our own categories to gives users additional info, afaik

File: starter-workflows/code-scanning/scorecards.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+    
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@v1
+
+      - name: "Run analysis"
+        # TODO: update with a hash once we have a release.
+        uses: ossf/scorecard-action@feat/rempol
```
Comment (laurentsimon) [inline starter-workflows/code-scanning/scorecards.yml:28]: cc @naveensrinivasan we're going to need to use the more official image from docker registry

Comment (naveensrinivasan) [reply starter-workflows/code-scanning/scorecards.yml:28]: Why docker registry? Is there a requirement?

Comment (laurentsimon) [reply starter-workflows/code-scanning/scorecards.yml:28]: my bad, I mean GitHub registry!

Comment (laurentsimon) [reply starter-workflows/code-scanning/scorecards.yml:28]: created this a few days back too keep track of it https://github.com/ossf/scorecard-action/issues/12. It should not change much from what you already did. Let me know on the issue if there's something else I'm missing.

File: starter-workflows/code-scanning/scorecards.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          #       * read:discussion
+          #
+          # After creation, copy the token.
+          #
+          # 2. Create an encrypted secret at https://github.com/<org>/<repo>/settings/secrets/actions
+          #    with the following settings:
+          #     - Name: SCORECARD_TOKEN
+          #     - Value: the value of the token created in step 1 above.
+          repo_token: ${{ secrets.SCORECARD_TOKEN }}
+          publish_results: true
```
Comment (laurentsimon) [inline starter-workflows/code-scanning/scorecards.yml:50]: @olivekl please comment on the comment :-)

**Comment by laurentsimon** [comment]
friendly ping. This is blocking the the submission of the workflow to GitHub, which in turn is blocking the verification of our action on the market place.

**Review by naveensrinivasan — APPROVED** [review]
Apologize for the delay. Thanks

File: starter-workflows/code-scanning/scorecards.yml (hunk 4 above)
Comment (olivekl) [reply starter-workflows/code-scanning/scorecards.yml:55]: 1) For consistency, should `read:discussion` say `write:discussion > read:discussion`?

2) "After creation, copy the token" --> change to "Create and copy the token." 

3) "Create an encrypted secret at..." --> change to "Create a new repository secret at..." (to match the wording on that page)

**Review by olivekl — APPROVED** [review]
LGTM, just suggestions for clarity

File: starter-workflows/code-scanning/scorecards.yml (hunk 4 above)
Comment (laurentsimon) [reply starter-workflows/code-scanning/scorecards.yml:55]: Thank you!

**Comment by laurentsimon** [comment]
Thanks everyone. Let's merge and we can re-visit. I want to get feedback from GitHub as well in the meantime

**Pull request merged**